docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,203 @@
+
+## Decision
+
+It would be speculative and inappropriate for EdgeX to select a unit of measure standard which is not widely adopted in the industry or choose a static unit of measure list that is incomplete with regard to possible IoT / edge use case needs.  At this time, there does not appear to be a single and unequivocal standard for units of measure that encompasses all EdgeX related use cases (now and in the future). 
+
+Therefore, EdgeX chooses not to select or adopt a unit of measure specification, standard, or code list to apply across the platform.  Instead, EdgeX adopters will be allowed to optionally specify which unit of measure specification, standard, or unit of measure code list they would like used in their instance(s) of EdgeX.
+
+### Specifying the Units of Measure
+
+Units of measure allowed by the instance of EdgeX will be specified in a configuration file (in TOML format).  This can either be specified directly in the core metadata configuration file (but then hard to change when using a containerized or snap form of the application) or in a separate and external configuration TOML file (making it easier for an adopter to provide units of measure even when the application is containerized or snapped up).  
+
+The units of measure in the configuration file can be attributed, optionally, to a specification, document, or other UoM definition source.  The `source` only helps provide the location of documentation about the origins and details of the units specified for the reader, but it will not be used or checked by EdgeX.  An optional default source can be provided at the top level configuration (as shown in the examples below) so that other sources are only needed when there are specific units used that are not found in the default source.
+
+The units of measure can be categorized for better organization and to allow for different sources to be specified for different units.  The categories are defined by the TOML section names (the UoM dot labels).
+
+**Sample TOML unit of measure configuration**
+
+``` toml
+[Uom]
+    Source="reference to source for all UoM if not specified below"
+    [Uom.temperature]
+    Source="www.weather.com"
+    Values="C,F,K"
+    [Uom.weights]
+    Source="www.usa.gov/federal-agencies/weights-and-measures-division"
+    Values="lbs,ounces,kilos,grams"
+```
+
+#### External UoM definition
+
+When the units of measure are specified in a separate (i.e. external definition) file, the location of the UoM file (specified by URI) will be specified in core metadata's configuration (currently in `res/configuration.toml`) - see example A below.  
+
+**Example A Metadata Configuration - external definition of UoM**
+``` toml
+[Writable]
+    [Writable.UoM]
+    Validation=false  ## false (meaning off) by default
+
+## in the non-writable area - example external file specified to units of measure
+[UoM]
+    UoMFile="file://tmp/edgex/metadata/uom.toml"  # the UoMFile location is specified as a URI.
```

Comment:
It may be useful to accept relative paths as well to be consistent with how tokenfile, profiles, devices paths are set. E.g. expect path to start with `./` or be a URI.

Reply:
@farshidtz - per monthly architect's meeting today 2/28/22, it was decided that file location would be specified as either a relative or absolute path specification.  URI concept was tabled until EdgeX 3.0.  Feeling was that it was adding too much complexity and additional scope to the first implementation of the UoM need at this time.